ares/component/processor/tlcs900h/instruction.cpp

```diff
@@ -630,172 +625,226 @@ auto TLCS900H::instructionRegister(R register) -> void {
     return instructionPop(register);
   }
   case 0x06: {  //CPL r
-    if constexpr(Long) return undefined();
-    prefetch(4);
-    return instructionComplement(register);
+    if constexpr(!Long) {
+      prefetch(4);
+      return instructionComplement(register);
+    }
+    return undefined();
```

Comment:
Why does Clang require inverting the `if` test?

Reply:
The early return wasn't short-circuiting anything at compile time. The compiler still wanted to follow the calls and instantiate templated functions with unsupported sizes. GCC was generating warnings here too; this cleanup applies to both it and clang.